Lab/src/test/java/AnimalTest.java

```diff
@@ -0,0 +1,38 @@
+package agh.ics.oop;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class AnimalTest {
+
+    @Test
+    public void testOptionsParser(){
+        OptionsParser o = new OptionsParser();
+        MoveDirection[] directions = o.parse(new String[]{"a","r","f","f","x","left","u","backward"});
+        assertArrayEquals(new MoveDirection[] {MoveDirection.RIGHT,MoveDirection.FORWARD,MoveDirection.FORWARD,MoveDirection.LEFT,MoveDirection.BACKWARD}, directions);
+
+        MoveDirection[] directions2 = o.parse(new String[]{"left","right","backward","forward","x","x","l","l","b","o","f","r"});
+        assertArrayEquals(new MoveDirection[] {MoveDirection.LEFT,MoveDirection.RIGHT,MoveDirection.BACKWARD,MoveDirection.FORWARD,MoveDirection.LEFT,MoveDirection.LEFT,MoveDirection.BACKWARD,MoveDirection.FORWARD,MoveDirection.RIGHT},directions2);
+    }
+
+    @Test
+    public void testAnimalMovement(){
+        Animal a = new Animal();
+
+        assertEquals("Północ, (2,2)",a.toString());
+
+        OptionsParser o = new OptionsParser();
+        MoveDirection[] moves = o.parse(new String[]{"f","b","b","b","b","b","l"});
+        for (MoveDirection move : moves) a.move(move);
+        assertEquals("Zachód, (2,0)",a.toString());
+
+        MoveDirection[] moves2 = o.parse(new String[]{"right","right","f","f","f","f","f","f"});
+        for (MoveDirection move : moves2) a.move(move);
+        assertEquals("Wschód, (4,0)", a.toString());
```

Comment:
ok, przy czym w tego typu teście sprawdzamy rezultat końcowy całej sekwencji - jeśli po drodze program popełniłby kilka sprzecznych ze sobą błędów i wyszłoby na końcu na plus to test dalej by przechodził.